components/function-controller/design/glossary_proposal.md

```diff
@@ -0,0 +1,105 @@
+
+### Project naming convention
+
+This section refers to the high-level architecture elements, namely to the main projects:
+
+- Serverless - the new naming convention for the `function-controller`. Serverless is responsible for running a Function on a k8s cluster. It can contain its own
+  CRD.
+- Serverless-operator - the new naming convention for `serverless-manager`. Serverless-operator installs and configures Serverless.
+- Kyma-Keda-operator - the operator which installs and configures [KEDA](https://keda.sh/).
+
+### Component naming convention
+
+This section refers to the Serverless components:
+
+- Controller - responsible for creating and configuring k8s resources to finally run a function on a cluster. It is responsible for the reconciliation of the Function CR.
+- Webhook - responsible for defaulting, validation, and conversion of the Function CR, mutating the external registry Secret, and reconciling certificates.
+
+Proposed naming convention:
+
+Deployment with the controller as the main responsibility:
+
+- ${component_name}-controller
+
+In the case of introducing a separate CRD and a separate deployment:
+
+- ${component_name}-{crd_name}-controller
+
+Deployment with webhook as main responsibility:
+
+- ${component_name}-webhook
+
+Deployment with both of those components:
+
+- ${component_name}
+
+I decided to go with the pure component name as it contains two responsibilities and from a technical perspective it's very similar to the component.
+It might be confusing, and I am open to other proposals.
+
+### Kubebuilder component naming convention
+
+Serverless uses Kubebuilder to build a controller and/or webhook. This section describes the naming convention of the most detailed project layer, namely Kubebuilder components.
+
+Looking at the [architecture](./assets/kubebuilder-architecture.png) diagram, you can see that a program consists of a **process** which includes a **manager**.
+The **manager** can include 2 components:
+
+- Controller, which focuses on the reconciliation of a given k8s resource. It uses predicates and the reconciler.
+- Webhook, which works with `AdmissionRequests`.
+
+Proposed naming convention:
+
+For controller reconcile loop inside the manager:
+
+- ${crd_name}-reconcile
+- ${component_name}-${crd_name}-reconcile
+
+For webhook inside the manager:
+
+- ${component_name}-validaton-webhook
+- ${component_name}-${crd_name}-validaton-webhook
+- ${crd_name}-validaton-webhook
+
+For serverless runtimes:
+
+- ${runtime_name}, eg.: `python310`
+
+## Summary
+
+The table lists the terms from the most general to the most detailed ones:
+
+| component name                | responsibility                                 |
+|-------------------------------|------------------------------------------------|
+| serverless                    | The product like `Keda`                        |
```

Comment:
```suggestion
| serverless                    | the product, such as Keda                      |
```

Reply:
I think it should be "KEDA". See their documentation: https://keda.sh